File: templates/python/zh/text/code/beeperLine.py

＃文件：BeeperLine.py
＃------------------------------
＃使用while循环放置一行锥体 s。
＃此程序适用于任何规模的世界。
from 卡雷尔.stanford import *

＃程序从main开始
def main():
   ＃重复直到karel面对墙
   while front_is_clear():
      ＃在当前正方形上放置一个锥体
      放锥体()
      ＃ move下一个方块
      move()
   ＃解决击剑事件
   放锥体()

File: templates/java/zh/text/code/beeperLine.java
/*
 * 文件：锥体行.java
 * ----------
 * 使用“while”循环放置锥体 s的行。该程序适用
 * 于任何规模的世界。
 */
import stanford.卡雷尔.*;
public class 锥体行 extends 卡雷尔 {
   public void run() {
      /* 重复，直到卡雷尔面临一堵墙 */
      while(frontIsClear()) {
         /* 在当前的广场上放置一个锥体 */ 
         putBeeper();
         /* move到下一个广场 */
         move();
      }
      /* 解决了围栏后的错误 */
      putBeeper();
   }
}